pilot/cmd/pilot-agent/main.go

```diff
@@ -100,25 +101,27 @@ var (
 			}
 
 			// set values from registry platform
-			if len(role.IPAddress) == 0 {
+			if len(proxyIP) != 0 {
+				role.IPAddresses = append(role.IPAddresses, proxyIP)
+			} else {
 				if registry == serviceregistry.KubernetesRegistry {
-					role.IPAddress = os.Getenv("INSTANCE_IP")
+					role.IPAddresses = append(role.IPAddresses, os.Getenv("INSTANCE_IP"))
```

Comment:
I think the logic below applies to all platforms. I would structure this as 
if instance_ip is set, use it.. else auto-detect proxy IPs. no need to have a check for registry type.